server/src/views/frame.js

```diff
@@ -6,21 +7,23 @@ exports.Frame = React.createClass({
       return <div>Not Found</div>;
     }
 
+    let shot = new AbstractShot(this.props.backend, this.props.id, this.props.shot);
+
     let favicon = "";
-    if (this.props.shot.favicon) {
-      favicon = <link rel="shortcut icon" href={this.props.shot.favicon} />;
+    if (shot.favicon) {
+      favicon = <link rel="shortcut icon" href={shot.favicon} />;
     }
 
     let snippet = "",
-      clipNames = this.props.shot.clipNames();
+      clipNames = shot.clipNames();
 
     if (clipNames.length) {
-      snippet = this.props.shot.getClip(clipNames[0]).image.url;
+      snippet = shot.getClip(clipNames[0]).image.url;
     }
 
     let linkTextShort = "";
-    if (this.props.shot.url) {
-      let txt = this.props.shot.url;
+    if (shot.url) {
+      let txt = shot.url;
       if (txt.length > 50) {
```

Comment:
That's correct. At the time I did this, I didn't want to use css because I didn't know how wide the window was going to be and the link was centered in an element that was as wide as the window.